### ISSUE TITLE
Feature Request: Graph verification without code generation

### ISSUE THREAD

**Issue by davidburstromspotify** [google/dagger#2075]
TL;DR; Running the Dagger annotation processor to verify the dependency graph but without generating code is useful for fast and safe iterations when using Dagger Reflect, called e.g. Dagger Verify.

When using https://github.com/JakeWharton/dagger-reflect (e.g. by using https://github.com/soundcloud/delect) for dependency injection, any errors in the DI graph will be discovered in runtime, which makes iteration tedious during development.

In a very large undisclosed project, the majority of the compilation phase is being spent compiling the generated Dagger sources:

DI solution | Annotation processing | Non-Dagger compilation | Dagger compilation
---------- | ---------------------- | ------------------------- | ---------------------
Dagger | 25% | 20% | 55%
Dagger Reflect | 0% | 100% | 0%
Dagger Verify |  57% | 43% | 0%

I haven't benchmarked how much time is being spent on the actual source generation, but it is likely non-negligible.

Compiling with Dagger Verify + Dagger Reflect would thus take 45% or less of a regular Dagger build, with the added benefit that graph errors are discovered statically.

**Comment by bcorso** [comment]
Hi David, 

I wonder how much "Dagger Verify + Dagger Reflect" would save you compared to just doing Dagger. While it's removing some compilation cost, it's also adding some run-time cost so the improvement in total time to build+run your test is unclear to me.

In your table, it would probably be helpful to compare time (e.g. in seconds) rather than percent so that we can compare tasks across rows.

**Comment by davidburstromspotify** [comment]
Hi Brad,

I can't disclose the actual numbers due to propriety, but to give you an idea, for regular Dagger the whole compilation phase is measured in minutes, not seconds. We're still working on getting a build up and running, but I have no reason to believe that Dagger Reflect will spend in the order of minutes setting up the runtime. The intention is that this will be used solely for dev builds, of course. Release builds will use regular Dagger for runtime performance.
